oshi-core/src/main/java/oshi/software/common/AbstractOperatingSystem.java

```diff
@@ -128,6 +131,11 @@ private int queryPlatformBitness() {
 
     protected abstract List<OSProcess> queryDescendantProcesses(int parentPid);
 
+    private static Set<Integer> getChildren(Map<Integer, Integer> parentPidMap, int parentPid) {
+        return parentPidMap.entrySet().stream().filter(e -> e.getValue().equals(parentPid))
```

Comment:
Might be useful to screen out the parent pid == pid case in this filter so you don't have to handle it separately later.